### PR TITLE
Stabilize IP associated constants

### LINKED ISSUE #44582: Tracking issue for IP constructors
Tracking issue for IP address convenience constructors, added in #44395, modified in https://github.com/rust-lang/rust/pull/52872.

```rust
impl Ipv4Addr {
    pub const LOCALHOST: Self = Ipv4Addr::new(127, 0, 0, 1);
    pub const UNSPECIFIED: Self = Ipv4Addr::new(0, 0, 0, 0);
    pub const BROADCAST: Self = Ipv4Addr::new(255, 255, 255, 255);
}

impl Ipv6Addr {
    pub const LOCALHOST: Self = Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1);
    pub const UNSPECIFIED: Self = Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 0);
}

### PR DESCRIPTION
**Pull request by tmccombs**
Fixes #44582

### CODE REVIEW

File: src/libstd/net/ip.rs [hunk 1]
```diff
@@ -357,47 +357,38 @@ impl Ipv4Addr {
...
-    /// #![feature(ip_constructors)]
     /// use std::net::Ipv4Addr;
     ///
     /// let addr = Ipv4Addr::LOCALHOST;
     /// assert_eq!(addr, Ipv4Addr::new(127, 0, 0, 1));
     /// ```
-    #[unstable(feature = "ip_constructors",
-               reason = "requires greater scrutiny before stabilization",
-               issue = "44582")]
+    #[stable(feature = "ip_constructors", since = "1.29.0")]
```
Comment (kennytm) [inline src/libstd/net/ip.rs]: The version should be 1.30.0

Comment (tmccombs) [reply src/libstd/net/ip.rs]: fixed. Is there an easy way to see what the current nightly version is?

Comment (faern) [reply src/libstd/net/ip.rs]: @tmccombs http://play.rust-lang.org/ will show the nightly version in use if you open the channel dropdown menu. But also, it's basically always current stable +2 versions.

**Comment by KodrAus** [comment]
Thanks @tmccombs!

@bors r+

**Comment by tmccombs** [comment]
The failed build looks like a hickup with accessing github.com

**Pull request merged**